plugins/ScheduledReports/Controller.php

```diff
@@ -167,10 +167,8 @@ public function unsubscribe()
 
         $view->reportName = $report['description'];
 
-        $nonce = Common::getRequestVar('nonce', '', 'string');
-
-        if (!empty($confirm) && Nonce::verifyNonce('Report.Unsubscribe', $nonce)) {
-            Nonce::discardNonce('Report.Unsubscribe');
+        if (!empty($confirm)) {
+            Nonce::checkNonce('Report.Unsubscribe');
```

Comment:
```suggestion
            try {
                Nonce::checkNonce('Report.Unsubscribe');
                $subscriptionModel->unsubscribe($token);
                $view->success = true;
            } catch (\Exception $e) {
                $view->nonce = Nonce::getNonce('Report.Unsubscribe');
            }
```

If the nonce is invalid, previous logic will generate a new nonce value from the Nonce::getNonce() function and assign that to the view in the else branch of the IF block.

If the nonce is invalid with the new logic, an Exception will be thrown in the Nonce::checkNonce() function because its not in a try catch block, the Exception will bubble up and break the control flow.

Reply:
Good catch, but actually I've done that on purpose. Currently if the nonce is considered invalid for any reason and you submit the form, the form will just reload. There are reasons beside a timed out nonce, that can trigger a failure. If that is the case, the form currently appears not working for the user, as there is no visual feedback. Now it will show an error message and the nonce is discarded. It might for sure be better to show that error within the form or similar, but I've considered this out of scope here. And it normally shouldn't happen anyway, not sure if it's worth the effort.